tests/test_modeling_common.py

```diff
@@ -741,68 +769,90 @@ def _create_and_check_torch_fx_tracing(self, config, inputs_dict, output_loss=Fa
+            if model.config.is_encoder_decoder:
+                model.config.use_cache = False  # FSTM still requires this hack -> FSTM should probably be refactored similar to BART afterward
+                labels = inputs.get("labels", None)
+                input_names = [
+                    "attention_mask",
+                    "decoder_attention_mask",
+                    "decoder_input_ids",
+                    "input_features",
+                    "input_ids",
+                    "input_values",
+                ]
+                if labels is not None:
+                    input_names.append("labels")
+
+                filtered_inputs = {k: v for (k, v) in inputs.items() if k in input_names}
+                input_names = list(filtered_inputs.keys())
+            else:
+                input_names = [
+                    "attention_mask",
+                    "bbox",
+                    "input_features",
+                    "input_ids",
+                    "input_values",
+                    "pixel_values",
+                    "token_type_ids",
+                    "visual_feats",
+                    "visual_pos",
+                ]
 
+                labels = inputs.get("labels", None)
+                start_positions = inputs.get("start_positions", None)
+                end_positions = inputs.get("end_positions", None)
+                if labels is not None:
+                    input_names.append("labels")
+                if start_positions is not None:
+                    input_names.append("start_positions")
+                if end_positions is not None:
+                    input_names.append("end_positions")
+
+                filtered_inputs = {k: v for (k, v) in inputs.items() if k in input_names}
+                input_names = list(filtered_inputs.keys())
+
+                if (
+                    isinstance(model, tuple(MODEL_FOR_SEQUENCE_CLASSIFICATION_MAPPING.values()))
+                    and not hasattr(model.config, "problem_type")
+                    or model.config.problem_type is None
+                ):
+                    model.config.problem_type = "single_label_classification"
+
+            model_output = model(**filtered_inputs)
+
+            # Note: `MKL_NUM_THREADS > 1` with `fork` will hang if the traced/scripted models call inputs.
+            # On CircleCI, it is set to `1`, so we are good.
+            start_methohd = "fork"
+            # `CUDA` works only with `spawn`.
+            if torch_device != "cpu":
+                start_methohd = "spawn"
+
+            ctx = torch.multiprocessing.get_context(start_methohd)
+
+            input_queue = ctx.Queue(1)
+            output_queue = ctx.JoinableQueue(1)
+
+            # To avoid the child process hanging on the line `traced_model = symbolic_trace(model, input_names)`.
+            # We will run `model.to(torch_device)` in the child process instead.
+            if torch_device != "cpu":
+                model.to("cpu")
```

Comment:
Don't know the exact reason, but passing model in CUDA via Queue will cause tracing issue. Let's pass it in CPU and send to CUDA device in the subprocess.

Reply:
You cannot really use multiprocessing with CUDA, they don't play along well together.